subsystems/networking/bluetooth_tethering_linux.go

```diff
@@ -0,0 +1,255 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	dbus "github.com/godbus/dbus/v5"
+	errw "github.com/pkg/errors"
+	"github.com/viamrobotics/agent/utils"
+	"go.viam.com/rdk/logging"
+)
+
+var errPairingRejected = &dbus.Error{Name: "org.bluez.Error.Rejected"}
+
+// this will be the bluetooth "agent" used for pairing requests.
+type basicAgent struct {
```

Comment:
is this name because we're modeling after the agent concept in bluez?

not a fan of the name basic agent. What about pairingAgent?